### DISCUSSION TITLE
number:[1-9] options for signcolumn

### DISCUSSION THREAD

**Discussion by trevorwhitney** [neovim/neovim#30500] [Q&A]
### Problem

I want the functionality of `signcolumn=number`, but I also want an additional column. I usually have between 0-2 signs per line. In the case of 2, 1 would be in the column to the left of the number, and the second would take the place of the number. In the case of 1, it would go in the spot to the left of the number (or it could take the place of the number, I don't really care much).

The problem currently is that if I want to conserve space by using `signcolumn=number`, I'm limited to seeing only 1 sign. I'm ok with having one additional column for signs, but currently if I want to be able to see 2 signs I need 3 column, 2 for signs and one for number. I'd like to accomplish this with 2 (by reusing the number column).

### Expected behavior

I can set `signcolumn=number:1`, which will give 2 columns for signs, the number column plus one additional.

**Comment by clason** [comment]
`:h 'statuscolumn'`

**Comment by trevorwhitney** [comment]
Maybe I can do this with `statuscolumn` but I can't quite figure out how? The logic needs to be dependent upon how many status symbols to show. ie. if `%s` = 2 show them both, otherwise show `%s%l`, maybe there's another variable I don't realize I have access to?

**Comment by luukvbaal** [comment]
The default 'statuscolumn' items will not allow you to to this but anything can be put in the statuscolumn area. The [statuscol.nvim](https://github.com/luukvbaal/statuscol.nvim) plugin can do this for you with segments like these (not exactly what you asked for but should be supported):


```lua
segments = {
  {text = {builtin.foldfunc}, click = "v:lua.ScFa"},
  {text = {builtin.lnumfunc}, click = "v:lua.ScLa", sign = {namespace = {"diagnostic"}}},
  {
    sign = {namespace = {"gitsigns"}, fillchar = "│", maxwidth = 1, colwidth = 1, wrap = true, foldclosed = true},
    click = "v:lua.ScSa",
  },
}
```
